lib/revision_table_access_check_filter.js

```diff
@@ -81,6 +81,13 @@ module.exports = function(hyper, req, next, options, specInfo) {
                 }
             });
         }
-        return next(hyper, req);
+        return next(hyper, req)
+        .then(function(res) {
+            if (req.query.redirect === false || mwUtil.isNoCacheRequest(req)) {
+                res.status = 200;
+                delete res.headers.location;
```

Comment:
@gwicke right, and I would take your argument as a +1 for leaving the location header in the response, but somehow I have a feeling you think we'd better remove it :)